src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/__tests__/ChartAxisConfiguration.test.tsx

```diff
@@ -0,0 +1,521 @@
+    const handleSubmit = jest.fn();
+
+    render(
+      <ChartBuilderFormsContextProvider initialForms={testFormState}>
+        <ChartAxisConfiguration
+          id="chartBuilder-major"
+          type="major"
+          configuration={testAxisConfiguration}
+          definition={verticalBarBlockDefinition}
+          data={testTable.results}
+          meta={testTable.subjectMeta}
+          onChange={noop}
+          onSubmit={handleSubmit}
+        />
+      </ChartBuilderFormsContextProvider>,
+    );
+
+    userEvent.clear(screen.getByLabelText('Size of axis (px)'));
+    await userEvent.type(screen.getByLabelText('Size of axis (px)'), '100');
+
+    userEvent.click(screen.getByRole('checkbox', { name: 'Sort ascending' }));
+
+    userEvent.click(screen.getByRole('button', { name: 'Save chart options' }));
+
+    await waitFor(() => {
+      const formValues: AxisConfiguration = {
+        dataSets: [],
+        groupBy: 'timePeriod',
+        min: 0,
+        max: undefined,
+        referenceLines: [],
+        showGrid: true,
+        size: 100,
+        sortAsc: false,
+        sortBy: 'name',
+        tickConfig: 'default',
+        tickSpacing: 1,
+        type: 'major',
+        visible: true,
+        unit: '',
+        label: {
+          text: '',
+        },
+      };
+      expect(handleSubmit).toHaveBeenCalledWith(formValues);
+    });
+  });
+
+  test('submitting succeeds with a group by a specific filter selected', async () => {
+    const handleSubmit = jest.fn();
+
+    render(
+      <ChartBuilderFormsContextProvider initialForms={testFormState}>
+        <ChartAxisConfiguration
+          id="chartBuilder-major"
+          type="major"
+          configuration={testAxisConfiguration}
+          definition={verticalBarBlockDefinition}
+          data={testTable.results}
+          meta={testTable.subjectMeta}
+          showGroupByFilter // EES-2467 remove this when BE done.
+          onChange={noop}
+          onSubmit={handleSubmit}
+        />
+      </ChartBuilderFormsContextProvider>,
+    );
+
+    userEvent.click(screen.getByRole('radio', { name: 'Filters' }));
+
+    fireEvent.change(screen.getByLabelText('Select a filter'), {
+      target: { value: 'school_type' },
+    });
+
+    userEvent.click(screen.getByRole('button', { name: 'Save chart options' }));
+
+    await waitFor(() => {
+      const formValues: AxisConfiguration = {
+        dataSets: [],
+        groupBy: 'filters',
+        groupByFilter: 'school_type',
+        min: 0,
+        max: undefined,
+        referenceLines: [],
+        showGrid: true,
+        size: 50,
+        sortAsc: true,
+        sortBy: 'name',
+        tickConfig: 'default',
+        tickSpacing: 1,
+        type: 'major',
+        visible: true,
+        unit: '',
+        label: {
+          text: '',
+        },
+      };
+      expect(handleSubmit).toHaveBeenCalledWith(formValues);
+    });
+  });
+
+  test('adding reference lines', async () => {
```

Comment:
Might want to group these related tests into their own `describe('reference lines')` block